scripts/us_census/acs5yr/subject_tables/common/acs_spec_validator.py

```diff
@@ -0,0 +1,849 @@
+      delimiter: delimiter seperating tokens within single column name string.
+    
+    Returns:
+      List of tokens that present in the spec but not in the columns from data source.
+  """
+    ret_list = []
+    # get list of unique tokens across all columns
+    token_list = get_tokens_list_from_column_list(column_name_list, delimiter)
+
+    ret_list = get_spec_token_list(spec_dict, delimiter)['token_list']
+
+    tokens_copy = ret_list.copy()
+
+    # ignore tokens beginning with an underscore or if token is a column name and appears in columnNameList
+    for token in tokens_copy:
+        if token.startswith('_'):
+            ret_list.remove(token)
+        elif token_in_list_ignore_case(token, token_list):
+            ret_list.remove(token)
+        if delimiter in token:
+            if token in column_name_list:
+                ret_list.remove(token)
+    return ret_list
+
+
+def find_columns_with_no_properties(column_name_list: list,
+                                    spec_dict: dict,
+                                    delimiter: str = '!!') -> list:
+    """Find all columns that do not assign any property a value
+    
+    Args:
+      column_name_list: List of all columns after dropping ignored columns.
+      spec_dict: Dict obj containing configurations for the import.
+      delimiter: delimiter seperating tokens within single column name string.
+    
+    Returns:
+      List of columns that do not assign any value to any property.
+  """
+    ret_list = []
+    for column_name in column_name_list:
+        no_prop_flag = True
+        # get token list of the column
+        for token in column_name.split(delimiter):
+            for prop in spec_dict['pvs'].keys():
+                if token_in_list_ignore_case(token,
+                                             spec_dict['pvs'][prop].keys()):
+                    # clear the flag when some property gets assigned a value
+                    no_prop_flag = False
+        # if the flag has remained set across all properties
+        if no_prop_flag:
+            ret_list.append(column_name)
+    return ret_list
+
+
+def find_ignore_conflicts(spec_dict: dict, delimiter: str = '!!') -> list:
+    """Find list of tokens that appear in ignoreColumn as well as a lookup for PV
+    NOTE: checks only tokens, ignores long column names
+
+    Args:
+      spec_dict: Dict obj containing configurations for the import.
+      delimiter: delimiter seperating tokens within single column name string.
+    
+    Returns:
+      List of tokens that are part of ignoreColumns as well as pv section.
+  """
+    ret_list = []
+
+    new_dict = copy.deepcopy(spec_dict)
+    new_dict.pop('ignoreColumns', None)
+    new_dict.pop('ignoreTokens', None)
+
+    spec_tokens = get_spec_token_list(new_dict, delimiter)['token_list']
+
+    if 'ignoreColumns' in spec_dict:
+        for ignore_token in spec_dict['ignoreColumns']:
+            if ignore_token in spec_tokens:
```

Comment:
Doesn't ignoreColumn contain the full column name with delimiters?
If so, should this condition check for any speck_token in ignore_tokens?
  set(delimiter.split(ignore_token)).intersection(set(spec_tokens))

Reply:
If entire column names are part of ignoreColumns, they will be ignored. This is specifically meant for token check only. Doing an intersection could lead to false positives as there might be one token which is in ignored column but also a pv lookup for another column that is not ignored, leading to false conflict warnings.

Reply:
I see. I missed that ignoreColumns could be tokens as well.